fontforge/python.c

```diff
@@ -14066,6 +14066,142 @@ return (NULL);
 return( Py_BuildValue("d",val));
 }
 
+static PyObject *PyFF_Font_get_style_set_names(PyFF_Font *self, void *UNUSED(closure)) {
+    int cnt, i;
+    SplineFont *sf = self->fv->sf;
+    struct otffeatname *fn;
+    struct otfname *on;
+    PyObject *ss_names_tuple, *ss_name_spec;
+    char *lang_str, tag_str[5];
+
+    for ( cnt=0, fn=sf->feat_names; fn!=NULL; fn=fn->next )
+        for ( on=fn->names; on!=NULL; on=on->next, ++cnt );
+    if ( cnt==0 )
+        return( PyTuple_New(0) );
+    ss_names_tuple = PyTuple_New(cnt);
+    for ( cnt=0, fn=sf->feat_names; fn!=NULL; fn=fn->next ) {
+        for ( on=fn->names; on!=NULL; on=on->next, ++cnt ) {
+            lang_str = copy(NOUI_MSLangString(on->lang));
+            for ( i=0; i<4; i++ )
+                tag_str[i] = (fn->tag>>(24-8*i)) & 0xFF;
+            tag_str[4] = '\0';
+            ss_name_spec = PyTuple_Pack(3, PyUnicode_FromString(lang_str), PyUnicode_FromString(tag_str), PyUnicode_FromString(on->name));
+            PyTuple_SetItem(ss_names_tuple, cnt, ss_name_spec);
+        }
+    }
+    return( ss_names_tuple );
+}
+
+static int PyFF_Font_set_style_set_names(PyFF_Font *self, PyObject *value, void *UNUSED(closure)) {
+    int i, j, rows, is_valid;
+    PyObject *ss_names_tuple;
+    char *lang_str, *tag_str, *name_str;
+    uint32_t tag, lang;
+    SplineFont *sf = self->fv->sf;
+    struct otffeatname *fn;
+    struct otfname *on;
+
+    if ( !PyTuple_Check(value) ) {
+        PyErr_Format(PyExc_TypeError, "Style set names must be set as a tuple." );
+        return( -1 );
+    }
+    rows = PyTuple_Size(value);
+    for ( i=0; i<rows; ++i ) {
+        ss_names_tuple = PyTuple_GetItem(value, i);
+        if ( !PyTuple_Check(ss_names_tuple) ) {
+            PyErr_Format(PyExc_TypeError, "Style set name specification must be a tuple." );
+            return( -1 );
+        }
+        if ( PyTuple_Size(ss_names_tuple)!=3 ) {
+            PyErr_Format(PyExc_TypeError, "Style set name specification must have 3 elements (language, style set and name)." );
+            return( -1 );
+        }
+
+        PyObject *lang_str_py = PyTuple_GetItem(ss_names_tuple, 0);
+        if ( !PyUnicode_Check(lang_str_py) ) {
+            PyErr_Format(PyExc_TypeError, "The first element of a style set specification must be a string." );
```

Comment:
The language should also be allowed to be specified as a number, as in the attribute `sfnt_names`.